code/game/objects/effects/weeds.dm

```diff
@@ -26,9 +26,16 @@
 
 
 /obj/effect/alien/weeds/Destroy()
+
 	if(parent_node)
 		SSweeds.add_weed(src)
 
+	if(loc)
+		for(var/obj/effect/alien/A in loc.contents)
+			if(A == src)
+				continue
+			qdel(A)
```

Comment:
Call `obj_destruction()` instead of directly deleting, for proper handling and long-term maintainability.